cpp/src/arrow/acero/CMakeLists.txt

```diff
@@ -19,6 +19,15 @@ add_custom_target(arrow_acero)
 
 arrow_install_all_headers("arrow/acero")
 
+macro(append_acero_avx2_src SRC)
+  if(ARROW_HAVE_RUNTIME_AVX2)
+    list(APPEND ARROW_ACERO_SRCS ${SRC})
```

Comment:
The old function was using `ACERO_SRCS`.  We could possibly also fix this by making the source list to modify an argument but I think it's fine to have it more self-contained this way.